src/basic/mountpoint-util.c

```diff
@@ -412,17 +408,14 @@ int path_get_mnt_id_at(int dir_fd, const char *path, int *ret) {
                   AT_NO_AUTOMOUNT |    /* don't trigger automounts, mnt_id is a local concept */
                   AT_STATX_DONT_SYNC,  /* don't go to the network, mnt_id is a local concept */
                   STATX_MNT_ID,
-                  &buf.sx) < 0) {
-                if (!ERRNO_IS_NOT_SUPPORTED(errno) && /* statx() is not supported by the kernel. */
-                    !ERRNO_IS_PRIVILEGE(errno) &&     /* maybe filtered by seccomp. */
-                    errno != EINVAL)                  /* glibc's fallback method returns EINVAL when AT_STATX_DONT_SYNC is set. */
+                  &sx) < 0) {
+                if (!ERRNO_IS_PRIVILEGE(errno)) /* maybe filtered by seccomp. */
```

Comment:
Is there any point in leaving this around still? You've dropped statx_fallback altogether which used to do a fstatat() fallback when statx is blocked. Now everything else that involves statx just fails hard, except for this one?